Properly identify Apple Silicon Macs.

This is a combination of three upstream commits:

https://hg.libsdl.org/SDL/rev/af22dd6c0787
https://hg.libsdl.org/SDL/rev/8a12de07fc37
https://hg.libsdl.org/SDL/rev/ac567246864a
--- configure.orig	2020-03-10 20:36:18.000000000 -0500
+++ configure	2020-09-04 00:03:44.000000000 -0500
@@ -21667,7 +21667,7 @@
           #import <Metal/Metal.h>
           #import <QuartzCore/CAMetalLayer.h>
 
-          #if !TARGET_CPU_X86_64
+          #if TARGET_CPU_X86
           #error Metal doesn't work on this configuration
           #endif
 
@@ -22570,7 +22570,7 @@
                   #include <Metal/Metal.h>
                   #include <QuartzCore/CAMetalLayer.h>
 
-                  #if !TARGET_CPU_X86_64
+                  #if TARGET_CPU_X86
                   #error Vulkan doesn't work on this configuration
                   #endif
 
@@ -24222,7 +24222,7 @@
     # so we'll just use libusb when it's available.
     case "$host" in
         # libusb does not support iOS
-        arm*-apple-darwin* | *-ios-* )
+        *-ios-* )
             skiplibusb=yes
             ;;
         # On the other hand, *BSD specifically uses libusb only
@@ -25072,7 +25072,7 @@
 fi
 
         ;;
-    arm*-apple-darwin*|*-ios-*)
+    *-ios-*)
         ARCH=ios
 
         CheckVisibilityHidden
